app/controllers/api/connections_controller.rb

```diff
@@ -122,10 +116,73 @@ def update_match
 
     end
 
+    def update_connection
+
+        if @connection.in_progress?
+
+            begin
+                @connection.class_status = params[:status]
+
+            rescue
+                return render_json_400("Updated status is invalid")
+            end
+
+            if @connection.in_progress?
+                return render_json_400("Updated status is invalid")
+            end
+
+            # Valid request
+
+            if @is_student
+                @connection.person_closed_connection = "student_closed"
+            else
+                @connection.person_closed_connection = "teacher_closed"
+            end
+
+            @connection.ended_at = Time.new
+
+            if @connection.save
+                # connection updated successfully, now we need to notify the other user
+
+                skill_class = @connection.match.skill_class
+
+                if @connection.given?
+                    text = "Connection to class to class #{skill_class.title} was closed. Class was given"
+                else
+                    text = "Connection to class to class #{skill_class.title} was closed. Class was canceled"
```

Comment:
"to class to class"

Also, eventually you should store all text on a locales file